### PR TITLE
docs: error handling

### PR DESCRIPTION
**Pull request by joaqcid**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ X ] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ X ] Documentation content changes
[ ] Other... Please describe:
```

Issue Number:

### CODE REVIEW

File: docs/advanced/errors.md [hunk 1]
```diff
@@ -19,10 +21,51 @@ export class MyErrorHandler implements ErrorHandler {
...
+```
+
+```TS
+  unhandled() {
+    this.store.dispatch(new UnhandledError()).pipe(
+      catchError(err => {
+        console.log("unhandled error on dispatch subscription")
+        return of('')
+      })
+    ).subscribe();;
```
Comment (splincode) [inline docs/advanced/errors.md]: ';;' ?

File: docs/advanced/errors.md [hunk 2]
```diff
@@ -10,6 +11,7 @@ export class MyErrorHandler implements ErrorHandler {
     console.log('ERROR! ', error);
 
     // Make sure to rethrow the error so Angular can pick it up
+    //
```
Comment (arturovt) [inline docs/advanced/errors.md]: is that needed? :)

File: docs/advanced/errors.md [hunk 3]
```diff
@@ -19,10 +21,51 @@ export class MyErrorHandler implements ErrorHandler {
...
-      provide: ErrorHandler, 
+      provide: ErrorHandler,
       useClass: MyErrorHandler
     }
   ]
 })
 export class AppModule { }
 ```
+
+## Handling errors within an @Action
```
Comment (arturovt) [inline docs/advanced/errors.md]: within an `@Action`

File: docs/advanced/errors.md [hunk 4]
```diff
@@ -19,10 +21,51 @@ export class MyErrorHandler implements ErrorHandler {
...
+  }
+```
+
+## Handling errors after dispatching an action
+
+If an unhandled exception is thrown inside an action, the error will be propagated to the `ErrorHandler` and you can also catch it subscribing to the `dispatch` Observable. If you subscribe to the `dispatch` Observable the error will be caught twice, once in the ErrorHandler and on your `dispatch` handle.
+
+```TS
+  @Action(UnhandledError)
+  unhandledError({ getState, setState }) {
```
Comment (arturovt) [inline docs/advanced/errors.md]: `ctx` instead of `{ getState, setState })`

File: docs/advanced/errors.md [hunk 5]
```diff
@@ -19,10 +21,51 @@ export class MyErrorHandler implements ErrorHandler {
...
+
+When you define an @Action you can handle error within the action and if you do so, the error will not propagate to Angular's global `ErrorHandler`, nor the `dispatch` Observable. This applies to both sync and async types of Actions.
+
+```TS
+  @Action(HandledError)
+  handledError({ getState, setState }) {
+    try {
+      // error is thrown
+    } catch (err) {
+      console.log("error catched inside @Action wont propagate to ErrorHandler or dispatch subscription")
```
Comment (arturovt) [inline docs/advanced/errors.md]: use single quotes

File: docs/advanced/errors.md (hunk 4 above)
Comment (markwhitfeld) [reply docs/advanced/errors.md]: Just for context, we prefer to not use destructuring too much in the docs because it assumes a level of knowledge from the reader. We aim for a little bit simpler syntax.

**Comment by markwhitfeld** [comment]
Thanks for the docs contribution!

File: docs/advanced/errors.md (hunk 4 above)
Comment (splincode) [reply docs/advanced/errors.md]: yep

```ts
  @Action(UnhandledError)
  unhandledError(ctx: StateContext<StateModel>) {
```

File: docs/advanced/errors.md [hunk 6]
```diff
@@ -2,6 +2,8 @@
 
 NGXS uses Angular's default `ErrorHandler` class, so if an action throws an error, Angular's `ErrorHandler` is called. You can easily override this flow by providing your own handler like so:
 
+NGXS propagates unhandled errors inside `@Action` to Angular's `ErrorHandler`. You can provide your own implementation of `ErrorHandler` and handle it the way you need like this:
```
Comment (markwhitfeld) [inline docs/advanced/errors.md]: @joaqcid I just merged and then saw this. Was this supposed to replace the previous paragraph?
You can do another PR if you need to fix it.

Comment (joaqcid) [reply docs/advanced/errors.md]: actually it was just a different option, just add it to see what you thought

Comment (joaqcid) [reply docs/advanced/errors.md]: i didnt mean to include it in the PR, i forgot to unstage it, let me know if you want me to revert it @markwhitfeld
